src/main/java/org/apache/ibatis/type/ArrayTypeHandler.java

```diff
@@ -38,19 +38,19 @@ public void setNonNullParameter(PreparedStatement ps, int i, Object parameter, J
   @Override
   public Object getNullableResult(ResultSet rs, String columnName) throws SQLException {
     Array array = rs.getArray(columnName);
-    return array == null ? null : array.getArray();
+    return (array == null || rs.wasNull()) ? null : array.getArray();
```

Comment:
@h3adache , Thanks for your comment. You're right! I've confirmed [the JDBC 4.2 specification](http://download.oracle.com/otn-pub/jcp/jdbc-4_2-mrel2-spec/jdbc4.2-fr-spec.pdf), it says about NULL values as follow:

```
15.2.3.3 Retrieving NULL values
The method wasNull can be called to determine if the last value retrieved was a
SQL NULL in the database.
When the column value in the database is SQL NULL, it may be returned to the Java
application as null, 0, or false, depending on the type of the column value.
Column values that map to Java Object types are returned as a Java null; those
that map to numeric types are returned as 0; those that map to a Java boolean are
returned as false. Therefore, it may be necessary to call the wasNull method to
determine whether the last value retrieved was a SQL NULL.
```
I will modify it at tonight(JST).